### ISSUE TITLE
Negative prompt confusion

### ISSUE THREAD

**Issue by zsogitbe** [ggerganov/llama.cpp#5709]
I have the strong feeling that the negative prompt (cfg_negative_prompt) is confused in the implementation with a positive prompt (guidance). Could you please check this in the code? Thank you.

A negative prompt is an instruction to exclude elements from generated content. That could be bad ethical content, objects, styles, colors, effects, etc.

**Comment by Solido** [comment]
The way to support negative prompts is not evident.

You need to first convert words to logits and then pass and negative weight to remove them from completion.

Server but not main benefits from a simpler solution that was not back ported by passing directly the words.

**Comment by zsogitbe** [comment]
> The way to support negative prompts is not evident.

Yes, but it is an important feature, for example, to control model ethics.
 
> You need to first convert words to logits and then pass and negative weight to remove them from completion.

Is this implemented somewhere?
 
> Server but not main benefits from a simpler solution that was not back ported by passing directly the words.

I am sorry, but I do not understand what you try to say here.

**Comment by Jeximo** [comment]
> I have the strong feeling that the negative prompt (cfg_negative_prompt) is confused in the implementation with a positive prompt (guidance).


More information is necessary..  Do you have an example?  What's your command line arguments?

**Comment by zsogitbe** [comment]
Thank you all. We managed to invert the weight and get a negative prompt instead of the positive one outside of llama.cpp. 
A comprehensive example about how to use a negative prompt would be here very useful. I am not sure that the current "negative prompt" implementation in llama.cpp with 'cfg_negative_prompt' is really a negative prompt, it is more probable that it is a positive prompt...
I will close this issue.
